tensorflow_addons/metrics/multilabel_metrics.py

```diff
@@ -0,0 +1,293 @@
+    Notes:
+        - The inputs `y_true` and `y_pred` have an assumed structure. Namely,
+            given a tensor of shape `(x, y, z)`, then `x` would correspond to
+            the batch size, `y` would correspond to the number of labels you
+            have and `z` would correspond to examples e.g. the following tensor
+            ```
+                [
+                    [
+                        [1, 1, 1, 0],
+                        [0, 1, 1, 0],
+                        [0, 0, 1, 0],
+                    ]
+                ]
+            ```
+            has a batch size of 1, three label classes and four examples, where
+            the first example has one label, the second two, the third three,
+            and fourth has zero labels.
+
+        - This metric works as well with batchless input i.e. shape `(y, z)`.
+        - Use of `activation` in this metric may cause unexpected effects if
+            using shape of 4D or if your input is structured differently.
+            In that case set `activation` to `None` and transform `y_pred`
+            prior to passing.
+    """
+
+    def __init__(
+        self,
+        name: str = "multi_label_macro_recall",
+        threshold: float = 0.5,
+        from_logits: bool = True,
+        activation: str = "sigmoid",
+        **kwargs
+    ):
+        """Creates a `MultiLabelMacroRecall` instance.
+
+        Args:
+            name: (Optional) string name of the metric instance. Defaults to
+                `"multi_label_macro_recall"`.
+            threshold: (Optional) float value to use to binarize labels and
+                predictions. Values greater than `threshold` will set to `1`.
+                Defaults to `0.5`.
+            from_logits: (Optional) boolean value to specifiy whether or not
+                `y_pred` are logits or have not yet been transformed via a final
+                activation function. If `from_logits` is `False`, `activation`
+                will be applied to `y_pred` prior to metric calculation. Defaults
+                to `True`.
+            activation: (Optional) string value of the activation function
+                to apply to `y_pred` if `from_logits=False`, e.g. `"sigmoid"`
+                [default] will apply the sigmoid function to `y_pred`.
+                Options include `["sigmoid"]`.
+        """
+        super(MultiLabelMacroRecall, self).__init__(name=name, **kwargs)
+        self._threshold = threshold
+        self._from_logits = from_logits
+        self._activation = activation
+        self._recall = self.add_weight(name="mlm_recall", initializer="zeros")
+
+        # NOTE: could be replaced  with tf confusion_matrix utils
+        # whether or not the overhead of that is worth is needs to be tested.
+        self._true_positives = self.add_weight(name="tp", initializer="zeros")
+        self._false_positives = self.add_weight(name="fp", initializer="zeros")
```

Comment:
Could we try using confusion_matrix for standardization

Reply:
That I'll leave for someone familiar with it. Other metrics in the addons do not use it either. I just know it exists.